CMakeLists.txt

```diff
@@ -105,7 +105,7 @@ endfunction()
 
 # OpenSSL libraries
 function(add_stratum_openssl_libs _TGT)
-    if(WITH_OPENSSL)
+    if(ES2K_TARGET)
```

Comment:
openssl::crypto library is not applicable for DPDK and Tofino targets? why?

Reply:
Because `tdi_fixed_function_manager.cc` contains a call to `OPENSSL_Cleanse()`, and ES2K is the only target that includes `tdi_fixed_function_manager.cc`.

Note that the PR consists of two commits. This change is part of the second commit, which undoes an incorrect edit in the first commit. The use of `OpenSSL::Crypto` is outside the scope of this PR.

That being said, I've found a better way of handling OpenSSL dependency, which I'll push in a separate PR.

Reply:
https://github.com/ipdk-io/stratum-dev/pull/144